js/src/utils/saveController.js

```diff
@@ -80,6 +80,48 @@
 
     },
 
+    getWindowObjectById: function(windowId) {
+      var returnObject = null;
+      jQuery.each(this.currentConfig.windowObjects, function(index, window) {
+        if (window.id === windowId) {
+          returnObject = window;
+          return false;
+        }
+      });
+      return returnObject;
+    },
+
+    getWindowAnnotationsList: function(windowId) {
+      if (this.windowsAnnotationsLists) {
+        return this.windowsAnnotationsLists[windowId];
+      } else {
+        return null;
+      }
+    },
+
+    getSlots: function() {
+      return this.slots;
+    },
+
+    getWindowElement: function(windowId) {
+      if (this.windowsElements) {
+        return this.windowsElements[windowId];
+      } else {
+        return null;
+      }
+    },
+
+    getStateProperty: function(prop) {
+      return this.get(prop, 'currentConfig');
+    },
+
+    get: function(prop, parent) {
+      if (parent) {
+        return this[parent][prop];
+      }
+      return this[prop];
+    },
+
```

Comment:
A bunch of helper functions to get at certain pieces of information in the global state.